nautobot_capacity_metrics/metrics.py

```diff
@@ -120,6 +125,36 @@ def metric_models(params):
     yield gauge
 
 
+def metric_versions():
+    """Return django, Nautobot, Python and app versions in Prometheus Metric format.
+
+    Return:
+        Iterator[GaugeMetricFamily]
+            nautobot_app_versions: the versions as labels
+    """
+    versions = {}
+    if PLUGIN_SETTINGS["versions"]["basic"]:
+        versions.update(
+            {"python": platform.python_version(), "django": django.get_version(), "nautobot": settings.VERSION}
+        )
+
+    # Collect app versions
+    if PLUGIN_SETTINGS["versions"]["plugins"]:
+        for app in settings.PLUGINS:
+            try:
+                app_module = importlib.import_module(app)
+            except ModuleNotFoundError:
+                logger.warning("Unable to find the python library %s", app)
+                continue
+            try:
+                versions[app] = app_module.__version__
+            except AttributeError:
+                logger.warning("Module %s does not have __version__ defined.", app)
+    gauge = GaugeMetricFamily("nautobot_app_versions", "Nautobot app versions", labels=versions.keys())
+    gauge.add_metric(versions.values(), 1)
```

Comment:
Is a gauge metric appropriate for version numbers? I can see that it would work for versions `1` or `1.0`, but it seems like it wouldn't be able to handle `1.0.0` and would be outright wrong for `1.10`?

Reply:
The actual version information is represented in the labels rather than the value of the metric. I found this is the best way to encode it as I wasn't able to find a satisfactory answer to encode version numbers into a Counter metric and neither Histogram nor Summary make any sense in my eyes.